llvm/lib/Target/WebAssembly/WebAssemblyInstrBulkMemory.td

```diff
@@ -51,25 +68,83 @@ defm DATA_DROP :
          [],
          "data.drop\t$seg", "data.drop\t$seg", 0x09>;
 
+}
+
+defm : BulkMemoryOps<I32, "32">;
+defm : BulkMemoryOps<I64, "64">;
+
+// Define copy/fill manually instead of using the `BulkMemoryOps` multiclass
+// because when a multiclass defines opcodes, it gives them anonymous names
+// and we need opcodes with names so that we can handle them with custom code.
```

Comment:
Not sure what this means. We use all opcodes in code by `WebAssembly::INSTRNAME` and most of them all inherit from some multiclass..? For example, this is from `LowerFPToInt`:
https://github.com/llvm/llvm-project/blob/27f66c9a84bc72acc2536013e08ea0a11161e81b/llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp#L481-L487
All instructions here inherit from some multiclass. For example, `ABS_F32` inherits from the multiclass `UNARY_FP`, and we can use their names fine:
https://github.com/llvm/llvm-project/blob/267be4a7f4ac69cfd1bec5223554bbe400c5636c/llvm/lib/Target/WebAssembly/WebAssemblyInstrFloat.td#L55

Am I missing something?

Reply:
Ah, it appears to be due to the `defm : BulkMemoryOps<...>`, which makes the opcodes "anonymous", and tablegen names their opcodes as `anonymous_9000MEMORY_COPY_A64` and `anonymous_9000MEMORY_COPY_A64` and `anonymous_9000MEMORY_FILL_A64`. If I change it to `defm : MEMORY_ : BulkMemoryOps<...>` and then remove the `MEMORY_` prefix from all the names, then they come out as just `MEMORY_COPY_A64` and `MEMORY_FILL_A64`.

I've now updated the code to use multiclasses for all instructions that have 32/64 variants.